feat(compass-collection): finish contents of mock data generator confirmation screen CLOUDP-333852

kpamaran opened the pull request:
<!--
  ^^^^^
  Please fill the title above according to https://www.conventionalcommits.org/en/v1.0.0/.

  type(scope): message <TICKET-NUMBER>

  eg. fix(crud): updates ace editor width in agg pipeline view COMPASS-1111

  NOTE: use `feat`, `fix` and `perf` for user facing changes that will be part of
  release notes.
-->
## Description
<!--- Describe your changes in detail -->
<!--- If applicable, describe (or illustrate) architecture flow -->
<!--- If the UI changes in a non-trivial way, consider adding screenshots/video illustrating the new flows -->
Jira: https://jira.mongodb.org/browse/CLOUDP-333852

The confirmation screen allows the user to preview and confirm to the contents sent to the LLM, which respects the setting for whether Generative AI features should send sample data.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] If this change updates the UI, screenshots/videos are added and a design review is requested
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it's updating a dependancy, link to the Pull Request that originally introduced the fix -->
- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Open Questions
<!--- Any particular areas you'd like reviewers to pay attention to? -->

## Dependents
<!--- If applicable, link PRs/commits that this PR is dependent on or is a dependency of. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

jcobis commented on packages/compass-collection/src/components/mock-data-generator-modal/raw-schema-confirmation-screen.tsx:
```diff
@@ -0,0 +1,131 @@
+    ? 'A sample of document values from your collection will be sent to an LLM for processing.'
+    : 'We have identified the following schema from your documents. This schema will be sent to an LLM for processing.';
+
+  // todo: should establish if unfinished schema analysis edge case should be handled within the modal or
```
Feel free to go ahead and do so. We can make the tooltip copy "Schema analysis in progress" for now

kpamaran commented on packages/compass-collection/src/components/mock-data-generator-modal/raw-schema-confirmation-screen.tsx:
```diff
@@ -0,0 +1,131 @@
+  // todo: should establish if unfinished schema analysis edge case should be handled within the modal or
+  // from the button opening the modal
+  const schemaAnalysisIncompletePlaceholder = (
+    <Body>Schema analysis has yet to complete successfully.</Body>
```
Updated. Given the change to disable the button instead this will be an unreachable state

jcobis replied:
It is unreachable now, no?

gribnoysup commented:
> You will notice that there is no syntax highlighting based on the mongo type in the schema-only variant. That may be viable with the customKeywords prop on the leafygreen Code component, but that requires 4 major version upgrades and may introduce breaking changes.

Any reason you are not using our code editor component in read only mode? It gives a better highlight, more performant, and visualizes large schemas in a more convenient way by collapsing most of the nested fields automatically. This is also more consistent with the rest of the application

kpamaran commented on packages/compass-collection/src/components/mock-data-generator-modal/raw-schema-confirmation-screen.tsx:
```diff
@@ -0,0 +1,117 @@
+          <Body className={descriptionStyles}>{descriptionText}</Body>
+          <Code language="javascript" copyable={false} className={codeStyles}>
+            {enableSampleDocumentPassing
+              ? JSON.stringify(schemaAnalysis.sampleDocument, null, 4)
```
Changed to the `Document` component. I don't think there should be any more representation issues on the schema-only or sample-doc variants

kpamaran commented:
@addaleax @gribnoysup @ncarbon I addressed latest round of diffs and updated the screenshots. There is a separate thread ([link](https://mongodb.slack.com/archives/C068FNVTTH7/p1757531531831249?thread_ts=1756905876.022039&cid=C068FNVTTH7)) for any additional design feedback from Maria, which I consider to be a non-blocker and can be followed up on.

jcobis commented on packages/compass-collection/src/transform-schema-to-field-info.ts:
```diff
@@ -221,3 +249,38 @@ function getMostFrequentType(types: SchemaType[]): SchemaType | null {
+      );
+    }
+
+    // Check that [] only appears as complete pairs and only at the end of the part
```
This may be overly stringent actually.

kpamaran replied:
Relating to this comment right? https://github.com/mongodb-js/compass/pull/7299#discussion_r2338059967

I'll remove for now

jcobis commented on packages/compass-collection/src/transform-schema-to-field-info.ts:
```diff
@@ -221,3 +228,36 @@ function getMostFrequentType(types: SchemaType[]): SchemaType | null {
+ * inputs are required to simulate these unlikely errors.
+ */
+function validateFieldPath(fieldPath: string) {
+  const parts = fieldPath.split('.');
```
This also has me thinking about fields whose names specifically end in `[]`. Another edge case to consider

jcobis replied:
Started a [Slack Thread](https://mongodb.slack.com/archives/C068FNVTTH7/p1757547604343399)

gribnoysup reviewed (approved):
Let's remove unneeded branching logic in the action, otherwise LGTM
